### DISCUSSION TITLE
A different patch2self question

### DISCUSSION THREAD

**Discussion by araikes** [dipy/dipy#2971] [Preprocessing]
Hi @ShreyasFadnavis,

This question is very different from my other question so I've separated it out a separate discussion. Apologies, because the initial post is very long.

### Question:
Total intracranial volume (TIV) and biological sex are generally colinear, often necessitating the inclusion of brain volume as a covariate in models (see e.g., https://doi.org/10.1002/hbm.26093). In a dataset I'm working on, `Patch2Self`, compared to MPPCA denoising, results in extensive correlations between regional DTI metrics and TIV (or, alternatively, demonstrates extensive sex differences). Given a potentially large difference in interpretation based on A) denoising method and B) inclusion of TIV as a covariate or not, is there a way to determine whether Patch2Self is _revealing_ a relationship in the data versus _creating_ a relationship? 

**T&DR**: If you scroll to the bottom and look at the figures, you can note that not only are there magnitude differences in sex difference effect sizes that are dependent on denoising method (effect sizes are much larger for Patch2Self) but also sign differences (particularly evident in models not including TIV; e.g. greater axial diffusivity in the GCC for males when using MPPCA and for the females when using Patch2Self).

### Background:
I'm working on a project looking at sex differences using single-shell DWI data. I processed the data using QSIprep (v. 0.19.1) and included `patch2self` as the denoising method. Following the ENIGMA DTI guidelines for TBSS, I obtained FA/MD/AD/RD estimates for the ROIs in the JHU DTI atlas. In exploring the data, I find that total intracranial volume is significantly correlated with all of the scalar types in the majority of the ROIs. See below for the correlations between TIV and Axial diffusivity.

```
 # Correlation Matrix (pearson-method)

Group     |    Parameter1 | Parameter2 |     r |         95% CI | t(61) |         p
-----------------------------------------------------------------------------------
ACR-L     | cat12_tiv_cm3 |      value | -0.27 | [-0.45, -0.08] | -2.76 | 0.007**  
ACR-R     | cat12_tiv_cm3 |      value | -0.16 | [-0.35,  0.04] | -1.55 | 0.124    
ALIC-L    | cat12_tiv_cm3 |      value | -0.24 | [-0.42, -0.04] | -2.38 | 0.019*   
ALIC-R    | cat12_tiv_cm3 |      value | -0.21 | [-0.39, -0.01] | -2.06 | 0.042*   
BCC       | cat12_tiv_cm3 |      value | -0.21 | [-0.39, -0.01] | -2.08 | 0.040*   
CGC-L     | cat12_tiv_cm3 |      value | -0.25 | [-0.43, -0.06] | -2.54 | 0.013*   
CGC-R     | cat12_tiv_cm3 |      value | -0.27 | [-0.44, -0.07] | -2.70 | 0.008**  
CGH-L     | cat12_tiv_cm3 |      value | -0.34 | [-0.50, -0.15] | -3.46 | < .001***
CGH-R     | cat12_tiv_cm3 |      value | -0.25 | [-0.43, -0.05] | -2.51 | 0.014*   
CP-L      | cat12_tiv_cm3 |      value | -0.37 | [-0.53, -0.19] | -3.89 | < .001***
CP-R      | cat12_tiv_cm3 |      value | -0.41 | [-0.56, -0.23] | -4.32 | < .001***
CST-L     | cat12_tiv_cm3 |      value | -0.26 | [-0.44, -0.07] | -2.64 | 0.010**  
CST-R     | cat12_tiv_cm3 |      value | -0.21 | [-0.39, -0.01] | -2.05 | 0.044*   
EC-L      | cat12_tiv_cm3 |      value | -0.22 | [-0.40, -0.02] | -2.21 | 0.030*   
EC-R      | cat12_tiv_cm3 |      value | -0.27 | [-0.44, -0.07] | -2.69 | 0.008**  
FX        | cat12_tiv_cm3 |      value |  0.15 | [-0.05,  0.34] |  1.50 | 0.138    
FX/ST-L   | cat12_tiv_cm3 |      value | -0.34 | [-0.51, -0.15] | -3.50 | < .001***
FX/ST-R   | cat12_tiv_cm3 |      value | -0.34 | [-0.51, -0.15] | -3.51 | < .001***
GCC       | cat12_tiv_cm3 |      value | -0.15 | [-0.34,  0.05] | -1.49 | 0.139    
ICP-L     | cat12_tiv_cm3 |      value | -0.29 | [-0.46, -0.09] | -2.94 | 0.004**  
ICP-R     | cat12_tiv_cm3 |      value | -0.30 | [-0.47, -0.11] | -3.04 | 0.003**  
IFO-L     | cat12_tiv_cm3 |      value | -0.16 | [-0.35,  0.04] | -1.61 | 0.110    
IFO-R     | cat12_tiv_cm3 |      value | -0.21 | [-0.40, -0.01] | -2.10 | 0.039*   
ML-L      | cat12_tiv_cm3 |      value | -0.34 | [-0.51, -0.15] | -3.50 | < .001***
ML-R      | cat12_tiv_cm3 |      value | -0.32 | [-0.49, -0.12] | -3.24 | 0.002**  
PCR-L     | cat12_tiv_cm3 |      value | -0.16 | [-0.35,  0.04] | -1.56 | 0.122    
PCR-R     | cat12_tiv_cm3 |      value | -0.26 | [-0.44, -0.07] | -2.65 | 0.009**  
PLIC-L    | cat12_tiv_cm3 |      value | -0.34 | [-0.51, -0.15] | -3.55 | < .001***
PLIC-R    | cat12_tiv_cm3 |      value | -0.38 | [-0.54, -0.19] | -3.99 | < .001***
PTR-L     | cat12_tiv_cm3 |      value | -0.12 | [-0.31,  0.09] | -1.12 | 0.264    
PTR-R     | cat12_tiv_cm3 |      value | -0.21 | [-0.39, -0.01] | -2.08 | 0.040*   
RLIC-L    | cat12_tiv_cm3 |      value | -0.23 | [-0.41, -0.03] | -2.24 | 0.027*   
RLIC-R    | cat12_tiv_cm3 |      value | -0.27 | [-0.45, -0.08] | -2.74 | 0.007**  
SCC       | cat12_tiv_cm3 |      value | -0.31 | [-0.48, -0.11] | -3.13 | 0.002**  
SCP-L     | cat12_tiv_cm3 |      value | -0.26 | [-0.44, -0.06] | -2.60 | 0.011*   
SCP-R     | cat12_tiv_cm3 |      value | -0.25 | [-0.43, -0.05] | -2.46 | 0.016*   
SCR-L     | cat12_tiv_cm3 |      value | -0.15 | [-0.34,  0.05] | -1.47 | 0.144    
SCR-R     | cat12_tiv_cm3 |      value | -0.21 | [-0.39, -0.01] | -2.09 | 0.040*   
SFO-L     | cat12_tiv_cm3 |      value | -0.22 | [-0.40, -0.02] | -2.17 | 0.032*   
SFO-R     | cat12_tiv_cm3 |      value | -0.17 | [-0.36,  0.03] | -1.70 | 0.092    
SLF-L     | cat12_tiv_cm3 |      value | -0.24 | [-0.42, -0.04] | -2.37 | 0.020*   
SLF-R     | cat12_tiv_cm3 |      value | -0.26 | [-0.44, -0.06] | -2.61 | 0.011*   
SS-L      | cat12_tiv_cm3 |      value | -0.26 | [-0.44, -0.06] | -2.59 | 0.011*   
SS-R      | cat12_tiv_cm3 |      value | -0.18 | [-0.36,  0.03] | -1.73 | 0.088    
UNC-L     | cat12_tiv_cm3 |      value | -0.05 | [-0.24,  0.16] | -0.44 | 0.660    
UNC-R     | cat12_tiv_cm3 |      value | -0.20 | [-0.38,  0.00] | -1.97 | 0.052  
```

### Problem
Motivated by A) my observation in the mouse data in the other discussion as well as B) the fact that including/not including TIV as a covariate in my model makes a massive difference in outcomes, I wanted to assess the impact of denoising strategy. So I ran _exactly the same code_ from start to finish (including QSIprep preprocessing, TBSS steps, and ENIGMA ROI extraction), but swapped out `patch2self` for `dwidenoise` in QSIprep. By contrast to Patch2Self, DTI scalars are not as extensively correlated with TIV. As can be seen below for axial diffusivity, the correlation values are smaller and statistically significant relationships are far more sparse.

```
# Correlation Matrix (pearson-method)

Group     |    Parameter1 | Parameter2 |        r |         95% CI |    t(61) |         p
-----------------------------------------------------------------------------------------
ACR-L     | cat12_tiv_cm3 |      value |     -0.16 | [-0.35, 0.04] |    -1.55 | 0.124  
ACR-R     | cat12_tiv_cm3 |      value |      0.01 | [-0.19, 0.21] |     0.11 | 0.914  
ALIC-L    | cat12_tiv_cm3 |      value |      0.22 | [ 0.02, 0.40] |     2.19 | 0.031* 
ALIC-R    | cat12_tiv_cm3 |      value |      0.22 | [ 0.02, 0.40] |     2.19 | 0.031* 
BCC       | cat12_tiv_cm3 |      value |      0.21 | [ 0.01, 0.40] |     2.12 | 0.037* 
CGC-L     | cat12_tiv_cm3 |      value |      0.04 | [-0.16, 0.24] |     0.39 | 0.697  
CGC-R     | cat12_tiv_cm3 |      value |      0.04 | [-0.16, 0.24] |     0.42 | 0.679  
CGH-L     | cat12_tiv_cm3 |      value |      0.06 | [-0.14, 0.25] |     0.56 | 0.578  
CGH-R     | cat12_tiv_cm3 |      value |      0.13 | [-0.07, 0.32] |     1.26 | 0.212  
CP-L      | cat12_tiv_cm3 |      value |     -0.04 | [-0.24, 0.16] |    -0.42 | 0.674  
CP-R      | cat12_tiv_cm3 |      value |     -0.13 | [-0.32, 0.07] |    -1.28 | 0.203  
CST-L     | cat12_tiv_cm3 |      value |      0.06 | [-0.15, 0.25] |     0.54 | 0.588  
CST-R     | cat12_tiv_cm3 |      value |      0.14 | [-0.06, 0.33] |     1.40 | 0.164  
EC-L      | cat12_tiv_cm3 |      value |      0.16 | [-0.04, 0.35] |     1.58 | 0.117  
EC-R      | cat12_tiv_cm3 |      value |  1.65e-03 | [-0.20, 0.20] |     0.02 | 0.987  
FX        | cat12_tiv_cm3 |      value |      0.26 | [ 0.07, 0.44] |     2.66 | 0.009**
FX/ST-L   | cat12_tiv_cm3 |      value |     -0.15 | [-0.34, 0.05] |    -1.50 | 0.138  
FX/ST-R   | cat12_tiv_cm3 |      value |     -0.06 | [-0.26, 0.14] |    -0.58 | 0.561  
GCC       | cat12_tiv_cm3 |      value |      0.28 | [ 0.09, 0.46] |     2.87 | 0.005**
ICP-L     | cat12_tiv_cm3 |      value |  1.30e-03 | [-0.20, 0.20] |     0.01 | 0.990  
ICP-R     | cat12_tiv_cm3 |      value |     -0.10 | [-0.30, 0.10] |    -0.98 | 0.329  
IFO-L     | cat12_tiv_cm3 |      value |      0.11 | [-0.10, 0.30] |     1.03 | 0.305  
IFO-R     | cat12_tiv_cm3 |      value |      0.05 | [-0.15, 0.25] |     0.47 | 0.642  
ML-L      | cat12_tiv_cm3 |      value |  9.01e-04 | [-0.20, 0.20] | 8.73e-03 | 0.993  
ML-R      | cat12_tiv_cm3 |      value |     -0.03 | [-0.23, 0.17] |    -0.28 | 0.782  
PCR-L     | cat12_tiv_cm3 |      value |      0.25 | [ 0.05, 0.43] |     2.45 | 0.016* 
PCR-R     | cat12_tiv_cm3 |      value |      0.06 | [-0.14, 0.26] |     0.58 | 0.560  
PLIC-L    | cat12_tiv_cm3 |      value |      0.10 | [-0.10, 0.29] |     0.97 | 0.333  
PLIC-R    | cat12_tiv_cm3 |      value | -8.06e-03 | [-0.21, 0.19] |    -0.08 | 0.938  
PTR-L     | cat12_tiv_cm3 |      value |      0.21 | [ 0.01, 0.40] |     2.10 | 0.038* 
PTR-R     | cat12_tiv_cm3 |      value |      0.02 | [-0.18, 0.22] |     0.20 | 0.841  
RLIC-L    | cat12_tiv_cm3 |      value |      0.16 | [-0.04, 0.35] |     1.61 | 0.111  
RLIC-R    | cat12_tiv_cm3 |      value |      0.09 | [-0.11, 0.29] |     0.91 | 0.366  
SCC       | cat12_tiv_cm3 |      value |      0.07 | [-0.14, 0.26] |     0.64 | 0.524  
SCP-L     | cat12_tiv_cm3 |      value |      0.09 | [-0.11, 0.29] |     0.91 | 0.365  
SCP-R     | cat12_tiv_cm3 |      value |      0.18 | [-0.02, 0.36] |     1.74 | 0.085  
SCR-L     | cat12_tiv_cm3 |      value |      0.27 | [ 0.07, 0.44] |     2.70 | 0.008**
SCR-R     | cat12_tiv_cm3 |      value |      0.15 | [-0.05, 0.34] |     1.47 | 0.146  
SFO-L     | cat12_tiv_cm3 |      value |      0.13 | [-0.07, 0.33] |     1.32 | 0.191  
SFO-R     | cat12_tiv_cm3 |      value |      0.10 | [-0.10, 0.29] |     0.95 | 0.344  
SLF-L     | cat12_tiv_cm3 |      value |      0.02 | [-0.18, 0.22] |     0.21 | 0.832  
SLF-R     | cat12_tiv_cm3 |      value |      0.03 | [-0.17, 0.23] |     0.28 | 0.784  
SS-L      | cat12_tiv_cm3 |      value |      0.08 | [-0.13, 0.27] |     0.74 | 0.464  
SS-R      | cat12_tiv_cm3 |      value |      0.22 | [ 0.02, 0.40] |     2.20 | 0.030* 
UNC-L     | cat12_tiv_cm3 |      value |      0.16 | [-0.04, 0.35] |     1.58 | 0.117  
UNC-R     | cat12_tiv_cm3 |      value |      0.08 | [-0.12, 0.28] |     0.81 | 0.420     
```

### Results
To demonstrate the effects of this, I regressed out age, education, mean framewise displacement and (optionally) TIV and then computed the sex difference effect size (Cohen's d) along with a permutation p-value and 95% bootstrap confidence intervals around the effect size estimate. Shown below are side-by-side comparison of MPPCA denoising and Patch2Self denoising on axial diffusivity. 

#### Without including TIV as a covariate
![image](https://github.com/dipy/dipy/assets/18663624/cb5f26a9-1d2c-4762-8fe7-e36dce11ae76)

Note that most regions show a large and statistically significant relationship after adjusting for multiple comparisons when using Patch2Self.

#### Including TIV as a covariate in the model
![image](https://github.com/dipy/dipy/assets/18663624/bdfac817-e33d-4db1-8e2e-235f80f377f1)

Note that fewer regions show a sex difference and none survive multiple comparisons correction when using Patch2Self. However, compared to MPPCA denoising, there are still multiple regions that exhibit moderate-to-large differences with confidence intervals that do not include 0.

**Comment by ShreyasFadnavis** [comment]
Hi @araikes -- this looks great!

qq: before I can make suggestions, what is the N (# of mice) here?

**Reply by araikes** [reply]
This is a subset with N = 64 people. Effects are similar in the larger sample (N = 214).

**Reply by ShreyasFadnavis** [reply]
I have been thinking about this for a while -- I think you can be sure that the denoising strategy is boosting your signal and not some spurious correlation. Even from the other discussion, since we are denoising each 4D volume separately in the N=64/214 sample, it seems unlikely that we induce bias of any sort. I have seen this with other work as well. For instance: https://www.sciencedirect.com/science/article/pii/S1361841523000282

**Reply by araikes** [reply]
How do you interpret, then, the following from that article:


> While patch2self indeed reduced the variability of estimates, experiments on synthetic data revealed that this comes at the cost of an additional bias, compared to when patch2self is not used, which becomes more relevant when including the spherical mean of the signal, as illustrated in supplementary Fig. S1. The reduction of the variability of estimates is particularly evident for the unbiased estimates, which we illustrated for the corpus callosum in supplementary Fig. S2.


Not trying to be nitpicky (because honestly I want to agree with you), but I'm trying to figure out how to be certain about the differences I observed that are directly attributable to the denoising method.

**Reply by ShreyasFadnavis** [reply]
No fair question-- that's a systematic error that has nothing to do with the algorithm! I'd reached out to the author about it. For simulated data, you need to change the parameters of P2S... but it was only after publication that I read this. 

In any case the above paper is on N=1 so the bias does not really make a difference, especially in your case.
